cmd/check_construction.go

```diff
@@ -163,7 +163,7 @@ func runCheckConstructionCmd(_ *cobra.Command, _ []string) error {
 	})
 
 	g.Go(func() error {
-		return tester.LogMemoryLoop(ctx)
+		return tester.LogMemoryLoop(ctx, constructionTester.GetLogger())
```

Comment:
This is not a good pattern - We should ideally initialize the tester with a logger rather than passing it an argument. That way it's always present to the tester package and we don't need to pass logger wherever we call `LogMemoryLoop`